### PR TITLE
[INFRA] Travis CI test gcc-9 on arm64

### LINKED ISSUE #1482: Infra: Enable arm64 CI builds
Now that arm64 is working, lets enable CI for it as well: https://docs.travis-ci.com/user/multi-cpu-architectures/

This can also be used for ppc64le & s390x

### PR DESCRIPTION
**Pull request by mr-c**
Fixes #1482

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -112,10 +127,12 @@ install:
     mkdir -p ${HOME_BIN_PATH}
   - |
     # install cmake
-    mkdir -p /tmp/cmake-download
-    wget --no-clobber --directory-prefix=/tmp/cmake-download/ https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
-    tar -C /tmp/ -zxvf /tmp/cmake-download/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
-    export PATH="/tmp/cmake-${CMAKE_VERSION}-Linux-x86_64/bin:${PATH}"
+    if [[ "${TRAVIS_CPU_ARCH}" != "amd64" ]]; then
```
Comment (mr-c) [inline .travis.yml]: cmake doesn't release arm64 binaries, but [Ubuntu Bionic has cmake 3.10.2](https://packages.ubuntu.com/bionic/cmake) which should be new enough

File: .travis.yml [hunk 2]
```diff
@@ -53,6 +53,21 @@ matrix:
...
       - CXXFLAGS="-std=c++2a"
       - CMAKE_VERSION=3.7.2
+  - os: linux
+    compiler: 'g++-9'
+    dist: bionic
+    addons:
+      apt:
+        sources:
+          - sourceline: ppa:ubuntu-toolchain-r/test
+        packages: ['g++-9', 'cmake']
```
Comment (mr-c) [inline .travis.yml]: I couldn't use the `<< : *linux-gcc-9` trick as I needed to customize the package list

File: .travis.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-sudo: required
```
Comment (mr-c) [inline .travis.yml]: `sudo: required` is deprecated and ignored

**Comment by marehr** [comment]
Thank you for the contributation. We currently limited our travis builds for per commit builds, to have a result after an hour. Arm builds should be nightly only.

**Comment by mr-c** [comment]
@marehr Okay. I don't have permissions to configure Travis cron jobs. The instructions are at https://docs.travis-ci.com/user/cron-jobs/ 

Some logic with `TRAVIS_EVENT_TYPE` will need to be added

**Comment by mr-c** [comment]
Huh:
```
/home/travis/build/seqan/seqan3/test/unit/alignment/pairwise/alignment_range_test.cpp:68:37: error: ‘constexpr bool testing::internal::InstantiateTypedTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TYPED_TEST_CASE_P is deprecated, please use INSTANTIATE_TYPED_TEST_SUITE_P [-Werror=deprecated-declarations]

   68 | INSTANTIATE_TYPED_TEST_CASE_P(alignment_range_iterator,

      |                                     ^~~~~~~~~~~~~~~~~~~                    

In file included from /home/travis/build/seqan/seqan3-build/vendor/googletest/googletest/include/gtest/gtest.h:62,

                 from /home/travis/build/seqan/seqan3/test/unit/alignment/pairwise/alignment_range_test.cpp:8:

/home/travis/build/seqan/seqan3-build/vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1230:16: note: declared here

 1230 | constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }

      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by marehr** [comment]
> Huh:
> 
> ```
> /home/travis/build/seqan/seqan3/test/unit/alignment/pairwise/alignment_range_test.cpp:68:37: error: ‘constexpr bool testing::internal::InstantiateTypedTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TYPED_TEST_CASE_P is deprecated, please use INSTANTIATE_TYPED_TEST_SUITE_P [-Werror=deprecated-declarations]
> 
>    68 | INSTANTIATE_TYPED_TEST_CASE_P(alignment_range_iterator,
> 
>       |                                     ^~~~~~~~~~~~~~~~~~~                    
> 
> In file included from /home/travis/build/seqan/seqan3-build/vendor/googletest/googletest/include/gtest/gtest.h:62,
> 
>                  from /home/travis/build/seqan/seqan3/test/unit/alignment/pairwise/alignment_range_test.cpp:8:
> 
> /home/travis/build/seqan/seqan3-build/vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:1230:16: note: declared here
> 
>  1230 | constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
> 
>       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Please rebase :)

**Comment by mr-c** [comment]
All tests pass. This is ready to be turned into a Travis CI cron build by a team member who has permissions to do so: https://docs.travis-ci.com/user/cron-jobs/

**Comment by marehr** [comment]
I'll close it for now and will later comeback for it when we switched to github actions. Thank you for giving me an example how to do this :)

**Pull request closed without merging**